src/server/src/modules/job.ts

```diff
@@ -0,0 +1,61 @@
+import { IJob, Job } from '@entities/job';
+import { getRepository } from 'typeorm';
+/**
+ * @description saves a new job in the database
+ * @param targetYears string[]
+ * @param hoursPerWeek number
+ * @param description string
+ * @param expirationDate Date
+ * @param startDate Date
+ * @param endDate Date
+ * @param type string[]
+ * @param title string
+ * @param status string
+ * @param minSalary number
+ * @param maxSalary number
+ * @param departmentId string
+ * @returns Promise
+ */
+export const createJob = (
+    targetYears: IJob['targetYears'],
+    hoursPerWeek: IJob['hoursPerWeek'],
+    description: IJob['description'],
+    expirationDate: IJob['expirationDate'],
+    startDate: IJob['startDate'],
+    endDate: IJob['endDate'],
+    type: IJob['type'],
+    title: IJob['title'],
+    status: IJob['status'],
+    minSalary: IJob['minSalary'],
+    maxSalary: IJob['maxSalary'],
+    departmentId: IJob['departmentId']
+) => {
+    if (expirationDate == null) {
+        // Set expiration date to 2 years after the current date.
+        expirationDate = new Date();
+        expirationDate.setFullYear(expirationDate.getFullYear() + 2);
+    }
+    if (endDate == null) {
+        // Set to expiration date
+        endDate = expirationDate;
+    }
+    if (maxSalary == null) {
+        maxSalary = minSalary;
+    }
+    const repository = getRepository(Job);
+    const jobToInsert = repository.create({
+        targetYears,
+        hoursPerWeek,
+        description,
+        expirationDate,
+        startDate,
```

Comment:
Here startDate comes from the body of the request which is going to be a string since the only types that are available in json are numbers, strings, and booleans. You have to convert startDate to a date object likes this new Date(startDate). Same for other date attributes

Reply:
On postman, when I specify startDate with string, it is automatically converted to a Date object. Do we still need the conversion? If so, do we want to change the type of startDate in IJob from Date to string?

Reply:
I checked your branch and startDate (and the other dates) come as strings from the request, but apparently, Postgres automatically converts them to date objects in the db. So yes, just update the type of date attributes to string in IJob. Lmk if that makes sense

Reply:
Done.